nemo_aligner/models/nlp/gpt/megatron_gpt_dpo_model.py

```diff
@@ -162,6 +183,38 @@ def fwd_output_and_loss_func(dataloader_iter, model, checkpoint_activations_all_
             else:
                 forward_args.pop("loss_mask")
 
+                if "cu_seqlens" in batch:  # packed sequence from DPOPackedDataset
+                    # these args are passed eventually into TEDotProductAttention.forward()
+                    cu_seqlens = batch["cu_seqlens"].squeeze()  # remove batch size dimension (mbs=1)
+
+                    max_seqlen = batch["max_seqlen"].squeeze() if "max_seqlen" in batch else None
+                    cu_seqlens_argmin = batch["cu_seqlens_argmin"] if "cu_seqlens_argmin" in batch else None
+
+                    # remove -1 "paddings" added in collate_fn
+                    if cu_seqlens_argmin is not None:
+                        cu_seqlens = cu_seqlens[: cu_seqlens_argmin.item()]
+                    else:
+                        cu_seqlens = cu_seqlens[: torch.argmin(cu_seqlens)]
+
+                    try:
+                        from megatron.core.packed_seq_params import PackedSeqParams
+                    except (ImportError, ModuleNotFoundError) as e:
+                        mcore_version = packaging.version.Version(version("megatron-core"))
+                        logging.error(
+                            f"megatron-core v{mcore_version} does not support training with packed sequence. "
+                            "Please use megatron-core >= 0.5.0, or set model.data.train_ds.packed_sequence=False"
```

Comment:
0.5.0 is pretty far back. do we need this check?

Reply:
I'll remove it. This check exists in nemo, but I don't think we need it here since aligner is much newer than nemo